src/core/hiwire.h

```diff
@@ -47,6 +47,20 @@ extern const JsRef Js_null;
 // For when the return value would be Option<JsRef>
 extern const JsRef Js_novalue;
 
+typedef struct Js_Identifier
+{
+  const char* string;
+  JsRef object;
+} Js_Identifier;
+
+#define Js_static_string_init(value)                                           \
+  {                                                                            \
+    .string = value, .object = NULL                                            \
+  }
+#define Js_static_string(varname, value)                                       \
+  static Js_Identifier varname = Js_static_string_init(value)
+#define Js_IDENTIFIER(varname) Js_static_string(JsId_##varname, #varname)
```

Comment:
Maybe you could add a comment to indicate that this is defined by analogy with https://github.com/python/cpython/blob/24da544014f78e6f1440d5ce5c2d14794a020340/Include/cpython/object.h#L17 to give a bit more context about why this is done for anyone reading it?